src/sonic-bgpcfgd/bgpcfgd/managers_static_rt.py

```diff
@@ -99,44 +102,55 @@ def split_key(key):
         else:
             return tuple(key.split('|', 1))
 
-    def static_route_commands(self, ip_nh_set, cur_nh_set, ip_prefix, vrf):
-        diff_set = ip_nh_set.symmetric_difference(cur_nh_set)
-
+    def static_route_commands(self, ip_nh_set, cur_nh_set, ip_prefix, vrf, route_tag, cur_route_tag):
         op_cmd_list = {}
-        for ip_nh in diff_set:
-            if ip_nh in cur_nh_set:
-                op = self.OP_DELETE
-            else:
-                op = self.OP_ADD
+        if route_tag != cur_route_tag:
+            for ip_nh in cur_nh_set:
+                op_cmds = op_cmd_list.setdefault(self.OP_DELETE, [])
+                op_cmds.append(self.generate_command(self.OP_DELETE, ip_nh, ip_prefix, vrf, cur_route_tag))
+            for ip_nh in ip_nh_set:
+                op_cmds = op_cmd_list.setdefault(self.OP_ADD, [])
+                op_cmds.append(self.generate_command(self.OP_ADD, ip_nh, ip_prefix, vrf, route_tag))
+        else:
```

Comment:
I was wondering if there are cases where both `route_tag` and `ip_nh_set` change at the same time? If that is the case, would this if-else check skip nh configuration?

Reply:
@shi-su we are deleting all the previous route/nexthop and creating based on new route/nexthop update. This should make sure we use latest nh configuration.